### PR TITLE
Fix for create database issue of managed instance.

### PR DESCRIPTION
**Pull request by ssreerama**
This PR adds fix for create database issue for Managed Instance.

ADS Issue: https://github.com/microsoft/azuredatastudio/issues/24559

CreateDB_MI

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs [hunk 1]
```diff
@@ -1772,6 +1772,11 @@ public DatabasePrototype(CDataContainer context)
                     // Azure instances don't have a model database we can query, so just use the defaults
                     this.originalState = new DatabaseData(context);
                 }
+                else if (context.Server.DatabaseEngineEdition == DatabaseEngineEdition.SqlManagedInstance)
+                {
+                    // SQl Managed Instance don't have a model database we can query, so just use the defaults to master
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs:1777]:
```suggestion
                    // SQL Managed Instances don't have a model database we can query, so just default to master
```

File: src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs [hunk 2]
```diff
@@ -1772,6 +1772,11 @@ public DatabasePrototype(CDataContainer context)
...
+                    this.originalState = new DatabaseData(context, "master");
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs:1778]: Is this what SSMS does?

@erinstellato-ms Does this seem like a reasonable approach? I can't remember if MI users will always be guaranteed to have access to master - do we want to consider either using the plain defaults like Azure does?

Comment (ssreerama) [reply src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs:1778]: I'm not sure how they handled in SSMS, but only approach that works in our case is by using the master database. I can check if we can use the default like azure.

Comment (erinstellato-ms) [reply src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs:1778]: @Charles-Gagnon @ssreerama  I'm checking, give me a few

Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs]: Also do we need to specify master, or can we just do the defaults like the Azure if-case directly above this?

**Review by corivera — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs (hunk 2 above)
Comment (Charles-Gagnon) [reply src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs]: It makes sense to try and use "reasonable" defaults when we can, but my question is whether master is actually going to represent reasonable defaults. If it isn't then yeah, I think your suggestion Cory is what I'd lean towards.

**Review by Charles-Gagnon — APPROVED** [review]
Using the defaults seems fine for now, but we should still follow up on what the "best" approach is here. Is there some set of reasonable defaults we can use for MI?

File: src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs (hunk 2 above)
Comment (ssreerama) [reply src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs]: It worked with or without specifying master as default. I'll check with Erin to confirm if we have any MI specific defaults or this approach is the correct one. Merging this one for now. Thank you all

**Pull request merged**